rest_tools/utils/auth.py

```diff
@@ -85,24 +85,27 @@ class OpenIDAuth:
     def __init__(self, url):
         self.url = url if url.endswith('/') else url+'/'
         self.public_keys = {}
+        self.provider_info = {}
         self.token_url = None
 
         self._refresh_keys()
 
     def _refresh_keys(self):
         try:
-            # discovery
-            r = requests.get(self.url+'.well-known/openid-configuration')
-            r.raise_for_status()
-            provider_info = r.json()
+            if not self.provider_info:
+                # discovery
+                r = requests.get(self.url+'.well-known/openid-configuration')
+                r.raise_for_status()
+                self.provider_info = r.json()
 
-            # get token url
-            self.token_url = provider_info['token_endpoint']
+                # get token url
+                self.token_url = self.provider_info['token_endpoint']
```

Comment:
More of a hunch than a bug, but this indentation seems off to me. I can't find `self.token_url` being set anywhere else.

Reply:
No, that's the right indentation.  It's meant to set the provider info and token url on the first attempt only.